sympy/simplify/ratsimp.py

```diff
@@ -159,6 +159,10 @@ def _ratsimpmodprime(a, b, allsol, N=0, D=0):
                         order=opt.order, polys=True)[1]
 
             S = Poly(r, gens=opt.gens).coeffs()
+
+            # TODO: Replace solve with linsolve, as of now solving for
```

Comment:
Currently, this is done by [`minsolve_linear_system`](https://github.com/sympy/sympy/blob/master/sympy/solvers/solvers.py#L1969) in `solvers.py` . I am not sure if  `linsolve` should implement this. Probably, the algorithm used here in `ratsimp.py` needs to be improved to use `linsolve`, as it doesn't looks clean to me.